### PR TITLE
Normalize config options main

### LINKED ISSUE #3473: Normalise all config options so default.ini is 100% commented out
The goal for `default.ini` and `local.ini` was always to have all values commented out.

It shouldn't be a hard effort to go through and make all of these values the correct defaults in the code.

Then, `default.ini` should just be **documentation** as to what the defaults are, so someone doesn't have to grep through a potentially unfriendly code base.

### PR DESCRIPTION
**Pull request by jiahuili430**
## Overview
Normalize some configuration options to comment out the default.ini file in the main branch.

## Related Issues or Pull Requests
This fixes #3473

## Checklist

- [x] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/chttpd/src/chttpd.erl [hunk 1]
```diff
@@ -148,7 +151,12 @@ start_link(Name, Options) ->
     end.
 
 get_server_options(Module) ->
-    ServerOptsCfg = config:get(Module, "server_options", "[]"),
+    ServerOptsCfg =
+        case Module of
+            "chttpd" -> config:get(Module,
```
Comment (iilyak) [inline src/chttpd/src/chttpd.erl:156]: strange indentation (we use `clause -> do_something` only if each clause fit on a single line).
```
case Module of
   "chttpd" ->
        config:get(Module, "server_options", ?DEFAULT_SERVER_OPTIONS);
    _ ->
        config:get(Module, "server_options", "[]")
end,
```

File: src/chttpd/src/chttpd_prefer_header.erl [hunk 2]
```diff
@@ -22,6 +22,11 @@
 -include_lib("couch/include/couch_db.hrl").
 
 
+-define(DEFAULT_PREFER_MINIMA,
```
Comment (iilyak) [inline src/chttpd/src/chttpd_prefer_header.erl:25]: Typo. It should be `DEFAULT_PREFER_MINIMAL`.

File: src/chttpd/src/chttpd.erl (hunk 1 above)
Comment (jiahuili430) [reply src/chttpd/src/chttpd.erl]: Changed.
Thanks for the review, @iilyak

File: src/chttpd/src/chttpd_prefer_header.erl [hunk 3]
```diff
@@ -47,8 +52,7 @@ filter_headers(Headers, IncludeList) ->
 
 
 get_header_list() ->
-    SectionStr = config:get("chttpd", "prefer_minimal", ""),
-    split_list(SectionStr).
+    split_list(config:get("chttpd", "prefer_minimal", ?DEFAULT_PREFER_MINIMA)).
```
Comment (iilyak) [inline src/chttpd/src/chttpd_prefer_header.erl:55]: typo

File: src/chttpd/src/chttpd_prefer_header.erl (hunk 2 above)
Comment (jiahuili430) [reply src/chttpd/src/chttpd_prefer_header.erl:25]: Changed.

File: src/chttpd/src/chttpd_prefer_header.erl (hunk 3 above)
Comment (jiahuili430) [reply src/chttpd/src/chttpd_prefer_header.erl:55]: Changed.

File: src/chttpd/test/eunit/chttpd_util_test.erl [hunk 4]
```diff
@@ -19,24 +19,24 @@
...
 setup() ->
     ok = config:set("httpd", "both_exist", "get_in_httpd", _Persist = false),
-    ok = config:set("chttpd", "both_exist", "get_in_chttpd", _Persist = false),
-    ok = config:set("httpd", "httpd_only", "true", _Persist = false),
-    ok = config:set("chttpd", "chttpd_only", "1", _Persist = false),
-    ok = config:set("couch_httpd_auth", "both_exist", "cha", _Persist = false),
-    ok = config:set("chttpd_auth", "both_exist", "ca", _Persist = false),
-    ok = config:set("couch_httpd_auth", "cha_only", "true", _Persist = false),
-    ok = config:set("chttpd_auth", "ca_only", "1", _Persist = false).
+    ok = config:set("chttpd", "both_exist", "get_in_chttpd", false),
```
Comment (iilyak) [inline src/chttpd/test/eunit/chttpd_util_test.erl:22]: Why remove `_Persist = `?

Comment (jiahuili430) [reply src/chttpd/test/eunit/chttpd_util_test.erl:22]: I ran `emilio` before and it gave a warning about `_Persist`.

Comment (jiahuili430) [reply src/chttpd/test/eunit/chttpd_util_test.erl]: Changed back.

**Review by iilyak — APPROVED** [review]
+1

**Comment by iilyak** [comment]
Very nice work. Thank you.

**Pull request merged**